paddle/fluid/operators/assert_op.cc

```diff
@@ -0,0 +1,110 @@
+/* Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License. */
+
+#include "paddle/fluid/framework/op_registry.h"
+#include "paddle/fluid/framework/var_type.h"
+#include "paddle/fluid/operators/controlflow/while_op_helper.h"
+#include "paddle/fluid/operators/tensor_formatter.h"
+
+const char kCondition[] = "CONDITION";
+const char kX[] = "X";
+const char kSummarizeNum[] = "summarize_num";
+
+namespace paddle {
+namespace operators {
+
+using framework::LoDTensor;
+
+class AssertOp : public framework::OperatorBase {
+ public:
+  AssertOp(const std::string &type, const framework::VariableNameMap &inputs,
+           const framework::VariableNameMap &outputs,
+           const framework::AttributeMap &attrs)
+      : OperatorBase(type, inputs, outputs, attrs) {}
+
+ private:
+  void RunImpl(const framework::Scope &scope,
+               const platform::Place &dev_place) const override {
+    const framework::Variable *cond_var_ptr = scope.FindVar(Input(kCondition));
+    PADDLE_ENFORCE_NOT_NULL(cond_var_ptr,
+                            platform::errors::NotFound(
+                                "Input(Condition) of AssertOp is not found."));
+    const LoDTensor &cond = cond_var_ptr->Get<LoDTensor>();
+    PADDLE_ENFORCE_EQ(
+        cond.dims(), paddle::framework::make_ddim({1}),
+        platform::errors::InvalidArgument(
+            "The shape of Input(Condition) of AssertOp must be 1. But now "
+            "the Condition's shape is ",
+            cond.dims().to_str(), ".\n"));
```

Comment:
It's better to put parentheses on the shape.
must be 1  ->  must be [1]
"the Condition's shape is ", cond.dims().to_str()  -> "the Condition's shape is [%s]", cond.dims().to_str()

Reply:
Sorry, I meant "the numel of Input(Condition) of AssertOp must be 1". I have changed it.

The returned string of DDim::to_str() has parentheses.

Reply:
> "the Condition's shape is ", cond.dims().to_str() -> "the Condition's shape is [%s]", cond.dims().to_str()

why not change this? error message is build by `Sprint`, the current writing may not work correctly

I think no need `\n`

Reply:
Sure, done